tensorflow/python/ops/image_ops_impl.py

```diff
@@ -3555,3 +3558,55 @@ def combined_non_max_suppression(boxes,
     return gen_image_ops.combined_non_max_suppression(
         boxes, scores, max_output_size_per_class, max_total_size, iou_threshold,
         score_threshold, pad_per_class)
+@tf_export('image.median_filter_2D')
+def median_filter_2D(input,filter_shape=(3,3)):
+    """  This methods takes 3D Tensor Images.
+         Other than Tensor it takes optional parameter filter_Size
+         Default Filter Shape = (3 , 3)
+         This Median Filtering is done by using 2D filters of user's choice
+         Filter_size should be odd
+         This method takes both kind of images where pixel values lie between 0 to 255 and where it lies between 0.0 and 1.0
+    """
+
+    input = image_ops_impl._Assert3DImage(input)
+    m,no,ch = int(input.shape[0]),int(input.shape[1]),int(input.shape[2])
+    filter_shapex = filter_shape[0]
+    filter_shapey = filter_shape[1]
+    if m < filter_shapex or no < filter_shapey:
```

Comment:
This static validation should only trigger if the shapes are known. To validate unknown / dynamic shapes use tf.shape and tf.assert

Reply:
I think solved as i have checked for Dimension(None) in the latest commit.

Reply:
filter_shapex and filter_shapey are also checked if they are integers or not.